quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusValidator.java

```diff
@@ -130,13 +130,61 @@ private void validatePropertyValue(String propertyName, ExtendedConfigDescriptio
 			return;
 		}
 
-		// TODO validate boolean, int, enums types
-		String type = metadata.getType();
+		String value = property.getValue().getText().trim();
+		if (value.length() == 0) {
+			return;
+		}
+		
+		String errorMessage = null;
+		if (isInvalidEnumValue(metadata, value)) {
+			errorMessage = "Invalid enum value: '" + property.getValue().getText().trim() + "' is invalid for type " + metadata.getType();
+		} else if (isValueTypeMismatch(metadata, value)) {
+			errorMessage = "Type mismatch: " + metadata.getType() + " expected";
+		}
+
+		if (errorMessage != null) {
+			addDiagnostic(errorMessage, property.getValue(), severity, ValidationType.value.name());
+		}
+	}
+
+	private boolean isInvalidEnumValue(ExtendedConfigDescriptionBuildItem metadata, String value) {
+
+		List<String> enums = metadata.getEnums();
+
+		if (enums == null) {
+			return false;
+		}
 
-		Boolean required = metadata.isRequired();
-		if (required != null && required) {
-			// TODO validate required property.
+		for (String e: enums) {
+			if (e.equals(value)) {
+				return false;
+			}
 		}
+
+		return true;
+	}
+
+	private boolean isValueTypeMismatch(ExtendedConfigDescriptionBuildItem metadata, String value) {
+		return !isBuildtimePlaceholder(value) &&
+		((metadata.isIntegerType() && !isIntegerString(value)) ||
+		(metadata.isFloatType() && !isFloatString(value)) ||
+		(metadata.isBooleanType() && !isBooleanString(value)));
+	}
+
+	private boolean isBooleanString(String str) {
+		return "true".equals(str) || "false".equals(str);
+	}
+
+	private boolean isIntegerString(String str) {
+		return str.matches("[+-]?\\d+");

```

Comment:
To improve performance it's better to precompile the regexp. See the following code:

```
	private static final Pattern INTEGER_PATTER = Pattern.compile("[+-]?\\d+");
....	
	private boolean isIntegerString(String str) {
		return INTEGER_PATTER.matcher(str).matches();
	}

```

Please do like this for other matches.

Reply:
BUT I think you should manage this method with Integer.parseInt loike https://stackoverflow.com/questions/237159/whats-the-best-way-to-check-if-a-string-represents-an-integer-in-java